### ISSUE TITLE
Feature request: put the fit time in evaluated_individuals_

### ISSUE THREAD

**Issue by louisabraham** [EpistasisLab/tpot#780]
It would be handy. GridSearchCV does it for example.
I also think I encountered some strange pipelines that did not stop after max_eval_time_mins, and this would help me to reproduce the issue.

**Comment by weixuanfu** [comment]
This is a old known issue for python.

The way CPython supports threading and asynchronous features has impacts on the accuracy of the timeout. For more background about this issue - that cannot be fixed - Please read Python gurus thoughts about Python threading, the GIL and context switching like these ones:

http://pymotw.com/2/threading/
https://wiki.python.org/moin/GlobalInterpreterLock

But I think it is a good idea to add this fit time into pipeline statistics.

**Comment by louisabraham** [comment]
I think the relevant code is there https://github.com/EpistasisLab/tpot/blob/507b45db01e8f88651f4ce8e03b607a5b50146f5/tpot/base.py#L1236-L1239

It seems you used the threading_timeoutable from stopit to handle the timeout. Why didn't you use instead the timeout parameter of [joblib.Parallel](https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html)?

**Comment by louisabraham** [comment]
Oh, the timeout parameter of joblib.Parallel raises a timeout if any task lasts to long.

Would https://github.com/joblib/joblib/pull/366 allow for a more precise time control?

**Comment by weixuanfu** [comment]
Maybe, I will look into it. But two issues need attentions when using timeout in joblib: 
1. TPOT uses `joblib` in `sklearn` to avoid adding one more dependency, so we need watch if scikit-learn updates the built-in joblib. 
2. this timeout in joblib only works when n_jobs !=1. We need a workaround for this.

**Comment by louisabraham** [comment]
1. Maybe we should just integrate some custom joblib code?
2. I'm not sure about what causes the issues with threads in the first place, but wouldn't multiprocessing.Process provide a timeout ability as well?
